src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,207 @@
+            prevNode.next = newNode;
+            size++;
+        }
     }
-
+    
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T element : list) {
+            add(element);
+        }
+        return true;
     }
-
+    
     @Override
     public T get(int index) {
-        return null;
+        checkIncludeIndex(index);
+        return getNode(index).value;
     }
-
+    
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIncludeIndex(index);
+        Node<T> currentNode = getNode(index);
+        T oldValue = currentNode.value;
+        currentNode.value = value;
+        return oldValue;
     }
-
+    
     @Override
     public T remove(int index) {
+        checkIncludeIndex(index);
+        if (size == 1 && index == 0) {
+            head = null;
+            tail = null;
+            size = 0;
+        }
+        int i = 0;
+        Node<T> currentNode = head;
+        while (currentNode != null) {
+            if (i == index) {
+                Node<T> nextNode = currentNode.next;
+                Node<T> prevNode = currentNode.prev;
+                if (prevNode == null) {
+                    nextNode.prev = null;
+                    head = nextNode;
+                    size--;
+                    return currentNode.value;
+                }
+                if (nextNode == null) {
+                    prevNode.next = null;
+                    tail = prevNode;
+                    size--;
+                    return currentNode.value;
+                }
+                nextNode.prev = prevNode;
+                prevNode.next = nextNode;
+                size--;
+                return currentNode.value;
+            }
+            i++;
+            currentNode = currentNode.next;
+        }
         return null;
     }
-
+    
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        while (currentNode != null) {
+            if (currentNode.value == object
+                    || (currentNode.value != null && currentNode.value.equals(object))) {
+                if (size == 1) {
+                    head = null;
+                    tail = null;
+                    size = 0;
+                    return true;
+                }
+                Node<T> nextNode = currentNode.next;
+                Node<T> prevNode = currentNode.prev;
+                if (prevNode == null) {
+                    nextNode.prev = null;
+                    head = nextNode;
+                    size--;
+                    return true;
+                }
+                if (nextNode == null) {
+                    prevNode.next = null;
+                    tail = prevNode;
+                    size--;
+                    return true;
+                }
+                nextNode.prev = prevNode;
+                prevNode.next = nextNode;
+                size--;
+                return true;
```

Comment:
let's move this to a separate method that will unlink neighbor nodes and use it here when the node is found